plugins/modules/aws_msk_cluster.py

```diff
@@ -0,0 +1,827 @@
+    purge_tags:
+        description: Remove tags not listed in I(tags) when tags is specified.
+        default: true
+        type: bool
+extends_documentation_fragment:
+    - amazon.aws.aws
+    - amazon.aws.ec2
+notes:
+    - All operations are time consuming, for example create takes 20-30 minutes,
+      update kafka version -- more than one hour, update configuration -- 10-15 minutes;
+    - Cluster's brokers get evenly distributed over a number of availability zones
+      that's equal to the number of subnets.
+"""
+
+EXAMPLES = r"""
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- aws_msk_cluster:
+    name: kafka-cluster
+    state: present
+    version: 2.6.1
+    nodes: 6
+    ebs_volume_size: "{{ aws_msk_options.ebs_volume_size }}"
+    subnets:
+      - subnet-e3b48ce7c25861eeb
+      - subnet-2990c8b25b07ddd43
+      - subnet-d9fbeaf46c54bfab6
+    wait: true
+    wait_timeout: 1800
+    configuration_arn: arn:aws:kafka:us-east-1:000000000001:configuration/kafka-cluster-configuration/aaaaaaaa-bbbb-4444-3333-ccccccccc-1
+    configuration_revision: 1
+
+- aws_msk_cluster:
+    name: kafka-cluster
+    state: absent
+"""
+
+RETURN = r"""
+# These are examples of possible return values, and in general should use other names for return values.
+
+bootstrap_broker_string:
+    description: A list of brokers that a client application can use to bootstrap.
+    type: complex
+    contains:
+        plain:
+            description: A string containing one or more hostname:port pairs.
+            type: str
+        tls:
+            description: A string containing one or more DNS names (or IP) and TLS port pairs.
+            type: str
+    returned: I(state=present) and cluster state is I(ACTIVE)
+cluster_info:
+    description: Description of the MSK cluster.
+    type: dict
+    returned: I(state=present)
+response:
+    description: The response from actual API call.
+    type: dict
+    returned: always
+    sample: {}
+"""
+
+import time
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    camel_dict_to_snake_dict,
+    compare_aws_tags,
+    AWSRetry,
+)
+
+
+@AWSRetry.jittered_backoff(retries=5, delay=5)
+def list_clusters_with_backoff(client, cluster_name):
+    paginator = client.get_paginator("list_clusters")
+    return paginator.paginate(ClusterNameFilter=cluster_name).build_full_result()
+
+
+@AWSRetry.jittered_backoff(retries=5, delay=5)
+def list_nodes_with_backoff(client, cluster_arn):
+    paginator = client.get_paginator("list_nodes")
+    return paginator.paginate(ClusterArn=cluster_arn).build_full_result()
+
+
+def find_cluster_by_name(client, module, cluster_name):
+    cluster_list = list_clusters_with_backoff(client, cluster_name).get("ClusterInfoList", [])
```

Comment:
While the _with_backup calls will automatically retry, they'll let non-temporary exceptions fall through.  BotoCoreException and ClientException should be caught here to avoid spitting big traces out at users.

Reply:
Fixed